### PR TITLE
chore(client): replace moxios with msw in brands

### PR DESCRIPTION
**Pull request by mindouro**
## Description
This replaces the usage of the `moxios` library with `msw` in the unit tests of the `brands`
chunk.

Refs #18 

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/client/src/brands/__fixtures__/getBrand.fixtures.ts [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
-      status: 404,
-    });
-  },
+  success: (response: Brand): RestHandler =>
+    rest.get(path, async (req, res, ctx) =>
+      res(ctx.status(200), ctx.json(response)),
+    ),
+  failure: (): RestHandler =>
+    rest.get(path, async (req, res, ctx) =>
+      res(ctx.status(404), ctx.json({ response: 'stub error' })),
```
Comment (nelsonleite) [inline packages/client/src/brands/__fixtures__/getBrand.fixtures.ts:13]:
```suggestion
      res(ctx.status(404), ctx.json({ message: 'stub error' })),
```

😇

File: packages/client/src/brands/__fixtures__/getBrands.fixtures.ts [hunk 2]
```diff
@@ -1,31 +1,15 @@
...
-      },
-    );
-  },
+  success: (response: Brands): RestHandler =>
+    rest.get(path, async (req, res, ctx) =>
+      res(ctx.status(200), ctx.json(response)),
+    ),
+  failure: (): RestHandler =>
+    rest.get(path, async (req, res, ctx) =>
+      res(ctx.status(404), ctx.json({ response: 'stub error' })),
```
Comment (nelsonleite) [inline packages/client/src/brands/__fixtures__/getBrands.fixtures.ts:13]:
```suggestion
      res(ctx.status(404), ctx.json({ message: 'stub error' })),
```

**Review by nelsonleite — CHANGES_REQUESTED** [review]

File: packages/client/src/brands/__fixtures__/getBrands.fixtures.ts (hunk 2 above)
Comment (mindouro) [reply packages/client/src/brands/__fixtures__/getBrands.fixtures.ts]: Fixed!

File: packages/client/src/brands/__tests__/__snapshots__/getBrand.test.ts.snap [hunk 3]
```diff
@@ -3,7 +3,8 @@
 exports[`brands client getBrand() should receive a client request error 1`] = `
 Object {
   "code": -1,
-  "message": "stub error",
+  "message": undefined,
```
Comment (anapim) [inline packages/client/src/brands/__tests__/__snapshots__/getBrand.test.ts.snap:6]: Is this correct? Is it missing a snapshot update?

File: packages/client/src/brands/__tests__/__snapshots__/getBrands.test.ts.snap [hunk 4]
```diff
@@ -3,7 +3,8 @@
 exports[`brands client getBrands() should receive a client request error 1`] = `
 Object {
   "code": -1,
-  "message": "stub error",
+  "message": undefined,
```
Comment (anapim) [inline packages/client/src/brands/__tests__/__snapshots__/getBrands.test.ts.snap:6]: Same here.

**Review by anapim — CHANGES_REQUESTED** [review]

**Review by nelsonleite — APPROVED** [review]

File: packages/client/src/brands/__tests__/__snapshots__/getBrand.test.ts.snap (hunk 3 above)
Comment (mindouro) [reply packages/client/src/brands/__tests__/__snapshots__/getBrand.test.ts.snap]: Related with Nelson's discussion. Fixed now :)

File: packages/client/src/brands/__tests__/__snapshots__/getBrands.test.ts.snap (hunk 4 above)
Comment (mindouro) [reply packages/client/src/brands/__tests__/__snapshots__/getBrands.test.ts.snap]: Related with Nelson's discussion. Fixed now :)

**Review by anapim — APPROVED** [review]

**Review by ricardojmf — APPROVED** [review]

**Pull request merged**